plugin/inlay_hint.py

```diff
@@ -112,40 +110,51 @@ def format_inlay_hint_tooltip(tooltip: Optional[Union[str, MarkupContent]]) -> s
 
 
 def format_inlay_hint_label(inlay_hint: InlayHint, session: Session, phantom_uuid: str) -> str:
+    tooltip = format_inlay_hint_tooltip(inlay_hint.get("tooltip"))
     result = ""
     can_resolve_inlay_hint = session.has_capability('inlayHintProvider.resolveProvider')
     label = inlay_hint['label']
     is_clickable = bool(inlay_hint.get('textEdits')) or can_resolve_inlay_hint
     if isinstance(label, str):
         if is_clickable:
-            inlay_hint_click_command = sublime.command_url('lsp_inlay_hint_click', {
-                'session_name': session.config.name,
-                'inlay_hint': cast(dict, inlay_hint),
-                'phantom_uuid': phantom_uuid
+            inlay_hint_click_command = sublime.command_url('lsp_on_double_click', {
+                'command': 'lsp_inlay_hint_click',
+                'args': {
+                    'session_name': session.config.name,
+                    'inlay_hint': cast(dict, inlay_hint),
+                    'phantom_uuid': phantom_uuid
+                }
             })
             result += '<a href="{command}">'.format(command=inlay_hint_click_command)
-        result += html.escape(label)
+        result += '<span title="{tooltip}">{value}</span>'.format(
+            tooltip=tooltip or 'Double-click to insert' if is_clickable else "",
```

Comment:
I'm not 100% sure, but I think this is not exactly right, because `or` has higher precedence than `if ... else` - https://docs.python.org/3/reference/expressions.html#operator-precedence

So this is currently the same as `(tooltip or 'Double-click to insert') if is_clickable else ""`, but we actually want `tooltip or ('Double-click to insert' if is_clickable else "")`

Because what we want (I think):
* if `is_clickable`, and server provides a tooltip, we show the tooltip
* if `is_clickable`, but server doesn't provide a tooltip, we show "Double-click to insert"
* if not clickable, but server provides a tooltip, we show the tooltip (perhaps a possible case for this would be something like label is `: bool` and tooltip is `Boolean value`)
* if not clickable, and server doesn't provide a tooltip, we show no tooltip at all

(but I have not tested with any language server, so please check whether what I wrote makes sense)

There is another similar place below, for the label parts.

You could also consider to not set the `title` attribute at all, if tooltip is `""` (empty). That would probably make the code a bit more complex, for example by building the result HTML string step by step. But feel free to ignore if `<span title="">` is fine.

Reply:
> So this is currently the same as (tooltip or 'Double-click to insert') if is_clickable else "", but we actually want tooltip or ('Double-click to insert' if is_clickable else "")

You are right.
will push a fix